components/CalendarPage/Calendar.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import { useState } from "react";
+import Calendar from "react-calendar";
+import "react-calendar/dist/Calendar.css";
+import styled from "styled-components";
+
+export default function CalendarFromReact({ entries }) {
+  const [date, setDate] = useState(new Date());
+  const [birthdays, setBirthdays] = useState([]);
+
+  const tileContent = ({ date, view }) => {
+    if (view === "month") {
+      const selectedDay = date.getDate();
+      const selectedMonth = date.getMonth() + 1;
+
+      const hasBirthday = entries.find((entry) => {
+        const birthDay = entry.birthday.split("-")[2];
+        const birthMonth = entry.birthday.split("-")[1];
+        return selectedDay == birthDay && selectedMonth == birthMonth;
+      });
+
+      if (hasBirthday) {
+        return <p>🎁</p>;
+      }
+    }
+  };
+
+  function handleClickDay(date, event, entries) {
+    setDate(date);
+
+    const selectedDay = date.getDate();
+    const selectedMonth = date.getMonth() + 1;
+
+    const birthdates = entries
+      .filter((entry) => {
+        const birthDay = entry.birthday.split("-")[2];
+        const birthMonth = entry.birthday.split("-")[1];
+        return birthDay == selectedDay && birthMonth == selectedMonth;
+      })
+      .map((entry, index) => (index ? ", " : "") + entry.name);
+    setBirthdays(birthdates);
+  }
+
+  return (
+    <>
+      <StyledCalendarContainer>
+        <Calendar
+          locale="de-DE"
+          tileContent={tileContent}
+          value={date}
+          onClickDay={(value, event) => handleClickDay(value, event, entries)}
+        />
+      </StyledCalendarContainer>
+      <StyledHeadline>Geburtstage an diesem Tag:</StyledHeadline>
+      <StyledNames>{birthdays}</StyledNames>
+    </>
+  );
+}
+
+const StyledCalendarContainer = styled.section`
+  margin: 8vw;
+  button {
+    margin: 2px;
+    background-color: #e6e6ea;
+    border-radius: 3px;
+  }
+  .react-calendar {
+    border: none;
+    border-radius: 4px;
+    padding: 3px;
+    box-shadow: 2px 2px 15px 2px #c4c4c4;
+  }
+  .react-calendar__month-view__days {
+    display: grid !important;
+    grid-template-columns: 14.2% 14.2% 14.2% 14.2% 14.2% 14.2% 14.2%;
+  }
+  .react-calendar__month-view__days__day--neighboringMonth {
+    opacity: 0.7;
+  }
+  .react-calendar__tile--active:enabled:hover {
+    background: #2ab7ca;
+  }
+  .react-calendar__year-view__months,
+  .react-calendar__decade-view__years,
+  .react-calendar__century-view__decades {
+    display: grid !important;
+    grid-template-columns: 20% 20% 20% 20% 20%;
+    &.react-calendar__year-view__months {
+      grid-template-columns: 33.3% 33.3% 33.3%;
+    }
+    .react-calendar__tile {
+      max-width: initial !important;
```

Comment:
I don't fully understand where these classes come from, are they styleable with classes? If it's possible it would be nice to avoid using styled components with classes, if this is the only way - so be it :D

Reply:
Yes, unfortunately this seems to be the only way to style this calendar... If I come across a better way, I will definitely do it. I had to use the DevTools to get these classes and style the different elements.